meson.build

```
@@ -44,6 +44,12 @@ else
     meson.add_dist_script(py, versioneer, '-o', '_version_meson.py')
 endif
 
+cc = meson.get_compiler('c')
+add_project_arguments(
+# https://github.com/cython/cython/issues/6870
+    cc.get_supported_arguments(['-Wno-error=unused-parameter']),
```

```suggestion
    cc.get_supported_arguments(['-Wno-error=unused-parameter', '-Wno-error=unused-function']),
```

This should resolve what is currently showing in CI.

There might be a few more. I recall looking at this a while back and being surprised at how many warnings were being suppressed in our current configuration (I think due to the meson-python pin?)

See also https://github.com/pandas-dev/pandas/pull/60681/files